python/triton/runtime/jit.py

```diff
@@ -589,10 +589,13 @@ def run(self, *args, grid, warmup, **kwargs):
         device = driver.active.get_current_device()
         stream = driver.active.get_current_stream(device)
         kwargs["debug"] = self.debug
+        param_kwargs = dict()
```

Comment:
It's going to add overhead, and is a concern when hooks are not used

Reply:
Considering the overhead when no hook is called, I already limit when there's a hook, then integrating `grid` and `warmup` into `kwargs`. If there's not, nothing happens.

Reply:
prefer literal `{}` instead of `dict`

Reply:
Already fix it